exporter/chronicleexporter/marshal.go

```diff
@@ -411,25 +412,61 @@ func (m *protoMarshaler) constructPayloads(rawLogs map[string][]*api.LogEntry, n
 				namespace = m.cfg.Namespace
 			}
 			ingestionLabels := ingestionLabelsMap[logType]
-			payloads = append(payloads, &api.BatchCreateLogsRequest{
-				Batch: &api.LogEntryBatch{
-					StartTime: timestamppb.New(m.startTime),
-					Entries:   entries,
-					LogType:   logType,
-					Source: &api.EventSource{
-						CollectorId: m.collectorID,
-						CustomerId:  m.customerID,
-						Labels:      ingestionLabels,
-						Namespace:   namespace,
-					},
-				},
-			})
+
+			request := m.buildGRPCRequest(entries, logType, namespace, ingestionLabels)
+
+			payloads = append(payloads, m.enforceMaximumsGRPCRequest(request)...)
 		}
 	}
 	return payloads
 }
 
-func (m *protoMarshaler) MarshalRawLogsForHTTP(ctx context.Context, ld plog.Logs) (map[string]*api.ImportLogsRequest, error) {
+func (m *protoMarshaler) enforceMaximumsGRPCRequest(request *api.BatchCreateLogsRequest) []*api.BatchCreateLogsRequest {
+	size := proto.Size(request)
+	if size > m.cfg.BatchRequestSizeLimitGRPC || len(request.Batch.Entries) > m.cfg.BatchLogCountLimitGRPC {
+		if len(request.Batch.Entries) < 2 {
+			m.teleSettings.Logger.Error("Single entry exceeds max request size. Dropping entry", zap.Int("size", size))
+			return []*api.BatchCreateLogsRequest{}
+		}
+
+		// split request into two
+		entries := request.Batch.Entries
+		mid := len(entries) / 2
+		leftHalf := entries[:mid]
+		rightHalf := entries[mid:]
+
+		request.Batch.Entries = leftHalf
+		otherHalfRequest := m.buildGRPCRequest(rightHalf, request.Batch.LogType, request.Batch.Source.Namespace, request.Batch.Source.Labels)
+
+		// re-enforce max size restriction on each half
+		enforcedRequest := m.enforceMaximumsGRPCRequest(request)
+		enforcedOtherHalfRequest := m.enforceMaximumsGRPCRequest(otherHalfRequest)
+
+		return append(enforcedRequest, enforcedOtherHalfRequest...)
+	}
+
+	return []*api.BatchCreateLogsRequest{
+		request,
+	}
```

Comment:
readability nit: in Go it's typical to handle the simple case and return early. This allows the code to stay flat and makes it easier to keep track of assumptions
```suggestion
	if size <= m.cfg.BatchRequestSizeLimitGRPC && len(request.Batch.Entries) <= m.cfg.BatchLogCountLimitGRPC {
		return []*api.BatchCreateLogsRequest{request}
	}
	
	if len(request.Batch.Entries) == 1 {
		m.teleSettings.Logger.Error("Single entry exceeds max request size. Dropping entry", zap.Int("size", size))
		return []*api.BatchCreateLogsRequest{}
	}

	// split request into two
	entries := request.Batch.Entries
	mid := len(entries) / 2
	leftHalf := entries[:mid]
	rightHalf := entries[mid:]

	request.Batch.Entries = leftHalf
	otherHalfRequest := m.buildGRPCRequest(rightHalf, request.Batch.LogType, request.Batch.Source.Namespace, request.Batch.Source.Labels)
	// re-enforce max size restriction on each half
	enforcedRequest := m.enforceMaximumsGRPCRequest(request)
	enforcedOtherHalfRequest := m.enforceMaximumsGRPCRequest(otherHalfRequest)

	return append(enforcedRequest, enforcedOtherHalfRequest...)
```